bridges/snowbridge/docs/v2.md

```diff
@@ -0,0 +1,310 @@
+
+The message is converted from $m_0$ to $x_0$ during message submission on BH. Dry-running submission will return $x_0$ to the relayer so that it can verify it is profitable.
+
+## Step 3: AH receives $x_0$ from BH
+
+Submitting the message $m_0$ will cause the following XCM, $x_0$, to be built on BH and dispatched to AH.
+
+```
+WithdrawAsset (KLT, 100)
+ReserveAssetDeposited(WETH, 2)
+PayFees (WETH, 1)
+SetAssetClaimer Bob            // derived from beneficiary
+AliasOrigin Ethereum/Alice     // derived from msg.sender
+InitiateAssetsTransfer asset=(KLT, 100) remoteFee=(WETH, 1) dest=KLT
+	DepositAsset (KLT, 100) beneficiary=Bob
+```
+
+```
+WithdrawAsset (KLT, 100)
+ReserveAssetDeposited(WETH, 2)
+PayFees (WETH, 1)
+SetAssetClaimer Ethereum/Alice   // derived from beneficiary
+AliasOrigin Ethereum/Alice       // derived from msg.sender
+InitiateAssetsTransfer asset=(KLT, 100) remoteFee=(WETH, 1) dest=KLT
+		DepositAsset (KLT, 100) beneficiary=Bob
+		Transact Bob.hello()
+```
+
+## Step 4: KILT Receives XCM from AH
+
+The following XCM $x_1$ is received from BH on AH.
+
+```
+*WithdrawAsset (KLT, 100)
+*ReserveAssetDeposited (WETH, 1)
+*PayFees (WETH, 1)
+*SetAssetClaimer Ethereum/Alice
+*AliasOrigin Ethereum/Alice   // origin preserved from AH
+SetAssetClaimer Bob
+DepositAsset (KLT, 100) beneficiary=Bob
+```
+
+```
+*WithdrawAsset (KLT, 100)
+*ReserveAssetDeposited (WETH, 1)
+*PayFees (WETH, 1)
+*SetAssetClaimer Ethereum/Alice
+*AliasOrigin Ethereum/Alice   // origin preserved from AH
+SetAssetClaimer Bob
+DepositAsset (KLT, 100) beneficiary=Bob
+Transact Bob.hello()          // executes with the origin from AH
+```
+
+# Relayer Rewards
+
+The tracking and disbursement of relayer rewards for both directions has been unified. Rewards in WETH are accumulated on BH but must be manually claimed to deposit them into an account.
+
+To claim, call following extrinsic, where $o$ is rewards account (origin), and $w$ is account on AH where the WETH will be paid out.
+
+$$
+\mathrm{claim}(o,w)
+$$
+
+For tax accounting purposes it might be desirable that $o \neq w$.
+
+# Top-Up
+
+Top-up of the relayer reward is viable to implement for either direction as extrinsics on Bridge Hub, respectively Ethereum.
+
+## Origin Preservation
+
+Origins for transact will be preserved by use of the `AliasOrigin` instruction. This instruction will have the following rules that parachain runtimes will need to allow:
+
+1. `AliasOrigin` can behave like `DescendOrigin`. This is allowed because it respects the hierarchy of multi-locations and does not allow jumping up.
+    1. Example location `Ethereum` can alias into `Ethereum/Alice` because we are descending in origin and this essentially is how the `DescendOrigin` instruction works.
+2. `AliasOrigin` must allow AH to alias into bridged locations such as `{ parents: 2, interior: GlobalConsensus(Ethereum) }` so that AH can act as a proxy for the bridge on parachains.
```

Comment:
```suggestion
2. `AliasOrigin` must allow AH to alias into bridged locations such as `{ parents: 2, interior: GlobalConsensus(Ethereum) }` and all of its internal locations so that AH can act as a proxy for the bridge on parachains.
```